tests/test_maudau.py

```diff
@@ -0,0 +1,127 @@
+import datetime as DT
+import json
+import os
+import pytest
+import tempfile
+from pyspark.sql import SparkSession
+from pyspark.sql.types import StructField, StructType, StringType
+from mozetl.maudau import maudau as M
+
+NARROW_SCHEMA = StructType([
+    StructField("client_id",             StringType(),  True),
+    StructField("submission_date_s3",    StringType(),  False),
+    StructField("subsession_start_date", StringType(), True)])
+
+generated = M.format_as_submission_date(M.today.isoformat())
+
+spark = (
+    SparkSession
+    .builder
+    .master("local")
+    # Uncomment to run while offline: from pyspark.conf import SparkConf
+    # .config(conf=SparkConf().set("spark.driver.host", "localhost"))
+    .appName("maudau")
+    .getOrCreate()
+)
+
+
+@pytest.fixture
+def make_frame():
+    cols = ['client_id', 'subsession_start_date', 'submission_date_s3']
+    values = [
+        ('a', '2017-05-01', '20170508'),
+        ('b', '2017-05-01', '20170501'),
+        ('c', '2017-05-01', '20170510'),
+        ('a', '2017-05-02', '20170503'),
+        ('b', '2017-05-03', '20170503'),
+        ('b', '2017-05-04', '20170511'),
+        ('a', '2017-05-05', '20170507'),
+    ]
+    rows = map(json.dumps, [dict(zip(cols, tup)) for tup in values])
+    jrdd = spark.sparkContext.parallelize(rows)
+    return spark.read.json(jrdd, NARROW_SCHEMA)
+
+
+def test_generate_filter_parameters():
+    expected0 = (
+        '2017-01-31', '2017-02-01', '20170131', '20170210'
+    )
+    actual0 = M.generate_filter_parameters(DT.date(2017, 1, 31), 0)
+    assert expected0 == actual0, str(actual0)
+
+    expected28 = (
+        '2017-01-03', '2017-02-01', '20170103', '20170210'
+    )
+    actual28 = M.generate_filter_parameters(DT.date(2017, 1, 31), 28)
+    assert expected28 == actual28
+
+
+def test_generate_counts():
```

Comment:
If these two test cases are exercising two different edge cases, it would be helpful to either comment on what each case is doing, or split them into more specifically-named functions.